modules/apps/frontend-taglib/frontend-taglib/src/main/java/com/liferay/frontend/taglib/servlet/taglib/ComponentTag.java

```diff
@@ -79,7 +79,8 @@ public String getModule() {
 			namespace = NPMResolvedPackageNameUtil.get(servletContext);
 		}
 		else {
-			namespace = NPMResolvedPackageNameUtil.get(request);
+			namespace = NPMResolvedPackageNameUtil.get(
+				pageContext.getServletContext());
```

Comment:
Makes sense, I think.

Reply:
This will cause modules to be looked up in the same OSGi bundle where the JSP invoking the tag is placed. Is this what we expect @jbalsas ? :thinking:

Reply:
I'm not sure about this... it could be this is always what we want, but could be that it's not... hard to tell...

We usually fix issues like this doing `servletContext="<%= application %>` where relevant. That's the reason we allow passing a different `servletContext`. This is the usual approach also followed in tags. Let's take this a closer look before pushing forward.

Reply:
I've looking at the other places where this kind of resolution takes place and it seems that we don't follow a pattern. These places are:

1. ComponentTag
2. DefineObjectsTag
3. BaseContainerTag
4. TemplateRendererTag
5. ReactRendererImpl

Some of them use the `pageContext`, others use the `request` and most of them allow overriding the value from the tag.

I think it might be a good idea to standardize this for the future, but that would be a breaking change, so it has to be done with care and making sure we don't break the portal.

So, I would apply the mechanism @jbalsas proposes now and maybe add this to the backlog @jbalsas ? :thinking:

Reply:
Yeah, there's probably even more places. A quick search of `git grep [-l] 'servletContext="<%= application %>' | wc -l` returns **670 occurrences in 375 files** so this is something that might take a while to untangle :)